### PR TITLE
Fix shape conv fuse opt

### PR DESCRIPTION
**Pull request by gedoensmax**
### Description

FIx:
- Multiples Convs into an Add+Relu will fuse the op although intermediates are needed @tianleiwu for viz
   image
- Also fixes an issue with Shape Initializers Merge as input, that occurs when the input initializer is the same across multiple nodes but not all nodes are Shape nodes. @er3x3 for viz

### CODE REVIEW

File: onnxruntime/core/optimizer/conv_activation_fusion.cc [hunk 1]
```diff
@@ -139,6 +154,10 @@ class ConvAddRelu : public NodeSelector {
     if (node_ep != kCudaExecutionProvider) {
       return std::nullopt;
     }
+    if (node.Name() == "Conv_3009")
```
Comment (tianleiwu) [inline onnxruntime/core/optimizer/conv_activation_fusion.cc:157]: Remove this debug code?

Comment (gedoensmax) [reply onnxruntime/core/optimizer/conv_activation_fusion.cc:157]: Oh sorry ! Completely overlooked that. I removed it.

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**